bundles/org.openhab.core.io.console/src/main/java/org/openhab/core/io/console/Completer.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.console;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Implementing this interface allows a {@link ConsoleCommandExtension} to
+ * provide completions for the user as they write commands.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+@NonNullByDefault
+public interface Completer {
```

Comment:
I would suggest to rename this class so the conflict with `org.apache.karaf.shell.api.console.Completer` is avoided. WDYT about `ConsoleCommandCompleter`?